content/inside-rust/call-for-testing-hint-mostly-unused.md

```diff
@@ -0,0 +1,207 @@
++++
+path = "inside-rust/2025/07/14/call-for-testing-hint-mostly-unused"
+title = "Call for Testing: Speeding up compilation with `hint-mostly-unused`"
+authors = ["Josh Triplett"]
++++
+
+I'm pleased to announce, and call for testing of, the nightly-only `rustc`
+`-Zhint-mostly-unused` option, and the corresponding nightly Cargo features
+`profile.hint-mostly-unused` and `hints.mostly-unused`. These options can help
+accelerate your Rust compile time in some cases, by avoiding compilation of
+items from your dependencies that you aren't using. Your feedback will help
+evaluate these features and make progress towards stabilizing them in the
+future.
+
+## Background
+
+When building a Rust library crate, the compiler generates compiled code for as
+much of the crate as it can (everything that isn't generic and isn't inlined),
+which gets linked into later crates into the dependency graph. However, some
+crates provide comprehensive APIs with a very large surface area, yet many of
+their users need only a few entry points. In such cases, the compiler currently
+spends time generating code for the entire crate, and the linker then ends up
+throwing most of that code away as unused.
+
+This can waste a substantial amount of compile time. Some large crates can take
+minutes to compile, and when you use these large crates as dependencies, they
+can take a disproportionate amount of the entire compilation time of your
+top-level crate.
```

Comment:
```suggestion
can take a disproportionate fraction of the entire compilation time for a full build.
```
Because adding `-Zhint-mostly-unused` will increase the compile time of the top-level crate, while lowering the compile time of the entire multi-crate build.